### PR TITLE
Add api endpoint for listing ApplicationDefinitions

### LINKED ISSUE #10294: List Endpoint for ApplicationDefinitions
**Note: This is split of into a separate story, because it is needed for KKP 2.21, whereas the other CRUD endpoints(#10035) are not**

Acceptance Criteria:

for now RBAC is not considered beyond standard cluster scoping (which is in line with current Kubermatic API design)
No special endpoint for getting the versions of an ApplicationDefinition (it will be done via standard GET)

### PR DESCRIPTION
**Pull request by SimonTheLeg**
Signed-off-by: Simon Bein <simontheleg@gmail.com>

**What does this PR do / Why do we need it**:

Add endpoint for listing ApplicationDefinitions

**Does this PR close any issues?**:
Fixes #10294 

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: pkg/handler/v2/routes_v2.go [hunk 1]
```diff
@@ -7272,6 +7278,30 @@ func (r Routing) updateApplicationInstallation() http.Handler {
...
+//       200: []ApplicationDefinition
+//       401: empty
+//       403: empty
+func (r Routing) listApplicationDefinitions() http.Handler {
+	return httptransport.NewServer(
+		endpoint.Chain(
+			middleware.TokenVerifier(r.tokenVerifiers, r.userProvider),
+			middleware.UserSaver(r.userProvider),
+		)(applicationdefinition.ListApplicationDefinitions(r.applicationDefinitionProvider)),
+		applicationdefinition.DecodeListApplicationInstallation,
```
Comment (lsviben) [inline pkg/handler/v2/routes_v2.go]:
```suggestion
		common.DecodeEmptyReq(),
```

File: pkg/handler/v2/application_definition/application_definition_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+				genKubermaticUser("John", "john@acme.com", true),
+			),
+			ExpectedHTTPStatus: http.StatusOK,
+			ExpectedAppDefs: []apiv2.ApplicationDefinition{
+				test.GenApiApplicationDefinition("appdef1"),
+				test.GenApiApplicationDefinition("appdef2"),
+			},
+		},
+		{
+			Name:            "admin can list all applicationdefinitions",
```
Comment (lsviben) [inline pkg/handler/v2/application_definition/application_definition_test.go]: I think you mixed up the test names :)

File: pkg/provider/kubernetes/application_definition.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	"k8c.io/kubermatic/v2/pkg/provider"
+
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// After refactoring the master-rbac controller, this provider can make use of the impersonated
+// master client (see 3ed4d3c3036cf6c4941d8703c2067fa655a20a74 for implementation).
+// However at the writing of this, it would require a large overhaul in the master-rbac-controller
+// for it to handle kubernetes cluster-scoped objects, that have no reference to kubermatic cluster
+// and projects. Therefore it was decided for now to make use of the master client directly for now.
```
Comment (lsviben) [inline pkg/provider/kubernetes/application_definition.go]:
```suggestion
// and projects. Therefore it was decided to make use of the master client directly for now.
```

File: pkg/handler/v2/application_definition/application_definition_test.go (hunk 2 above)
Comment (SimonTheLeg) [reply pkg/handler/v2/application_definition/application_definition_test.go]: oh yes! :D

**Review by vgramer — COMMENTED** [review]
/lgtm

**Pull request merged**